### PR TITLE
Feature/storage market provider

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**

### CODE REVIEW

File: core/vm/actor/builtin/v0/miner/miner_actor_state.cpp [hunk 1]
```diff
@@ -16,8 +17,10 @@ namespace fc::vm::actor::builtin::v0::miner {
 
   outcome::result<types::miner::MinerInfo> MinerActorState::getInfo(
       IpldPtr ipld) const {
-    OUTCOME_TRY(info0, ipld->getCbor<MinerInfo>(miner_info));
-    return std::move(info0);
+    OUTCOME_TRY(info, ipld->getCbor<MinerInfo>(miner_info));
+    OUTCOME_TRYA(info.window_post_proof_type,
```
Comment (wer1st) [inline core/vm/actor/builtin/v0/miner/miner_actor_state.cpp:21]: Why do you not satisfied by value from ipld?

Comment (Alexey-N-Chernyshov) [reply core/vm/actor/builtin/v0/miner/miner_actor_state.cpp:21]: RegisteredPoStProof type combines WinningPost and WindowPost and in v0 and v2 actors the WinningPost stored in IPLD miner state in 'window_post_proof_type'. But the function expects WindowPost, so Lotus determines WindowPost for SealProofType. There are much more code for actor in Lotus, see (https://github.com/filecoin-project/lotus/blob/master/chain/actors/builtin/miner/v0.go#L315). I don't know the reasoning for such things.

Comment (wer1st) [reply core/vm/actor/builtin/v0/miner/miner_actor_state.cpp:21]: Ok

**Review by wer1st — APPROVED** [review]

File: core/vm/actor/builtin/states/verified_registry_actor_state.hpp [hunk 2]
```diff
@@ -20,6 +21,18 @@ namespace fc::vm::actor::builtin::states {
...
     adt::Map<DataCap, adt::AddressKeyer> verified_clients;
+
+    /**
+     * @param address must be an id address
+     * @return data cap for the given address
+     */
+    outcome::result<boost::optional<DataCap>> getVerifiedClientDataCap(
+        const Address &address) {
+      if (!address.isId()) {
+        ERROR_TEXT("Can only look up ID addresses");
```
Comment (turuslan) [inline core/vm/actor/builtin/states/verified_registry_actor_state.hpp:32]:
```suggestion
        return ERROR_TEXT("Can only look up ID addresses");
```

File: core/api/make.cpp [hunk 3]
```diff
@@ -357,8 +374,33 @@ namespace fc::api {
...
         }};
-    // TODO(turuslan): FIL-165 implement method
-    api->MarketReserveFunds = {};
+
+    api->MarketReserveFunds = {[=](const Address &wallet,
+                                   const Address &address,
+                                   const TokenAmount &amount)
+                                   -> outcome::result<boost::optional<CID>> {
+      // TODO(a.chernyshov): method should use fund manager to ensure there is
+      // no reserved funds in market actor. Rework method after the fund manager
```
Comment (turuslan) [inline core/api/make.cpp:383]:
```suggestion
      // TODO(a.chernyshov): method should use fund manager to batch reserve and release requests.
      // Rework method after the fund manager
```

**Review by turuslan — APPROVED** [review]

**Pull request merged**